src/main/java/pl/edu/mimuw/logic/Solver.java

```diff
@@ -0,0 +1,53 @@
+package pl.edu.mimuw.logic;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Solver {
+
+  public ArrayList<FormulaVariable> variables;
```

Comment:
z jakiego powodu to jest publiczne? 😞  btw w polach klasowych też cześciej stodujmy typy interfejsow, np `List<FormulaVariable>`